### ISSUE TITLE
Color hex codes for segments given color seed

### ISSUE THREAD

**Issue by austinhoag** [google/neuroglancer#220]
Is there a straightforward way to extract the color hex codes, e.g. `#F9F607` for a segment ID or list of segment IDs given a color seed? This would be helpful for labeling segments in figures produced from Neuroglancer screenshots in which many segments are present.

**Comment by jbms** [comment]
There isn't at the moment, but that would be a good thing to add to the Python API.  Would you care to implement that, e.g. based on the code in src/neuroglancer/segment_color.ts?

**Comment by austinhoag** [comment]
I am interested in adding this to the Python API as well. I had a look at `src/neuroglancer/segment_color.ts` at one point since this appears to be where this mapping takes place. I have to be honest that I am still learning Typescript and it was not clear to me how to reproduce some of the Typescript code in Python. I will dig through and see what I don't understand, and maybe with some help I can reproduce the code in Python.

**Comment by jbms** [comment]
Great, feel free to ask if you have any questions.  There is also a GLSL implementation if you find that easier.  One way to be able to see the entire GLSL code at once is to set DEBUG_SHADER = true  in src/neuroglancer/webgl/shader.ts, then modify the shader code in segment_color.ts to contain a syntax error.  Then when you run Neuroglancer you will get a popup window showing you the full shader code.

**Comment by austinhoag** [comment]
I think as long as I can reproduce the murmurhash in `src/neuroglancer/gpu_hash/hash_function.ts` in Python the rest should not be too bad. I am having trouble reproducing this in Python however. I first tried a sklearn library: https://scikit-learn.org/stable/modules/generated/sklearn.utils.murmurhash3_32.html, but testing with a seed of 0, value of 59 I get a different result from your javascript implementation: ` 789950352` compared with the Python library: `1897138774`. 

I also attempted reproducing the `hashCombine()` function directly in Python, but I do not understand what the following lines mean:
```
value >>>= 0;
```
I was not able to find a reference for the `>>>=` operator.

And
```
value = Math.imul(value, k1) >>> 0;
```
I was not able to reproduce this with Python 3.7's operator.imul() function.

**Comment by jbms** [comment]
The hash function was based on murmurhash but excludes some rounds to be a bit faster (which also makes it a worse hash function, but that doesn't matter for this purpose), so you can't just use an existing murmurhash implementation.

In javascript, >>> is unsigned right shift, and >>> 0 simply coerces a value to a 32-bit unsigned integer, effectively computes the non-negative result mod 2^32.  (Normally numbers in javascript are represented as 64-bit floating point.)

Math.imul(value, k1) >>> 0 is simply doing a normal 32-bit unsigned integer multiply (with modular behavior on overflow).  In contrast, `value * k1` does a floating point multiply and doesn't have the correct overflow behavior.

In Python, normal integers are represented as arbitrary width numbers --- to get a 32-bit result, you can use: `& 0xffffffff`

For example, `value = (value * k1) & 0xffffffff`  in place of `value = Math.imul(value, k1) >>> 0`

**Comment by austinhoag** [comment]
Addressed in https://github.com/google/neuroglancer/pull/224
